### PR TITLE
fix: throw if given an unknown file ext

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
Addresses #5

File: src/lib/index.ts [hunk 1]
```diff
@@ -98,7 +100,7 @@ export class GoogleToken {
     if (!this.key && this.keyFile) {
       const mimeType = mime.getType(this.keyFile);
       if (mimeType === 'application/json') {
```
Comment (jinwoo) [inline src/lib/index.ts]: Suggestion. Isn't it better to use `switch` instead of `if-else-if` chain?

Comment (JustinBeckwith) [reply src/lib/index.ts]: Why not :)  Switched it up.

File: src/lib/index.ts [hunk 2]
```diff
@@ -97,39 +99,49 @@ export class GoogleToken {
...
-          this.key = key;
-        } else {
-          // assume .pem key otherwise
-          const key = await readFile(this.keyFile, 'utf8');
-          this.key = key;
-        }
+      switch (mimeType) {
+        case 'application/json':
+          // *.json file
+          const key1 = await readFile(this.keyFile, 'utf8');
```
Comment (jinwoo) [inline src/lib/index.ts]: Nit. If you embrace the `case` statement with parens, you can avoid naming local vars like `key1`, `key2`, etc.

```typescript
case 'application/json': {
  const key = ...
  ...
  break;
}
case 'application/x-x509-ca-cert': {
  const key = ...
  ...
}
...
```

File: src/lib/index.ts [hunk 3]
```diff
@@ -80,7 +80,9 @@ export class GoogleToken {
           .then(t => {
             callback(null, t);
           })
-          .catch(callback);
+          .catch(e => {
+            callback(e);
+          });
```
Comment (ofrobots) [inline src/lib/index.ts]: Why is this change necessary?

File: src/lib/index.ts (hunk 2 above)
Comment (ofrobots) [inline src/lib/index.ts]: nit: use a block scope inside each case statement and avoid variables with ordinal suffices (`key` vs `key1`).

File: src/lib/index.ts [hunk 4]
```diff
@@ -97,39 +99,49 @@ export class GoogleToken {
...
+          if (!this.key || !this.iss) {
+            const e = new Error('private_key and client_email are required.');
+            (e as NodeJS.ErrnoException).code = 'MISSING_CREDENTIALS';
+            throw e;
+          }
+          break;
+        case 'application/x-x509-ca-cert':
+          // *.pem file
+          this.ensureEmail();
+          const key2 = await readFile(this.keyFile, 'utf8');
```
Comment (ofrobots) [inline src/lib/index.ts]: nit: assign directly to `this.key`. Don't need this variable.

File: src/lib/index.ts [hunk 5]
```diff
@@ -97,39 +99,49 @@ export class GoogleToken {
...
+          this.key = key2;
+          break;
+        case 'application/x-pkcs12':
+          // *.p12 file
+          this.ensureEmail();
+          const key3 = await toPem(this.keyFile);
```
Comment (ofrobots) [inline src/lib/index.ts]: ditto

File: src/lib/index.ts [hunk 6]
```diff
@@ -97,39 +99,49 @@ export class GoogleToken {
...
-        }
+      switch (mimeType) {
+        case 'application/json':
+          // *.json file
+          const key1 = await readFile(this.keyFile, 'utf8');
+          const body = JSON.parse(key1);
+          this.key = body.private_key;
+          this.iss = body.client_email;
+          if (!this.key || !this.iss) {
+            const e = new Error('private_key and client_email are required.');
```
Comment (ofrobots) [inline src/lib/index.ts]: nit: Since this is repeated 3 times, make an custom Error class and use constructor.

File: src/lib/index.ts (hunk 2 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: Thank you.  Hadn't had my coffee yet, and was wondering why it was complaining about the scope.

File: src/lib/index.ts (hunk 3 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: Brain left head.  Had it there for debugging, fixed.

File: src/lib/index.ts (hunk 2 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: Fixed!

File: src/lib/index.ts (hunk 4 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: fixed

File: src/lib/index.ts (hunk 5 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: fixed

File: src/lib/index.ts (hunk 6 above)
Comment (JustinBeckwith) [reply src/lib/index.ts]: fixed
